pymc3/distributions/continuous.py

```diff
@@ -3786,6 +3765,39 @@ def logp(value, mu, sigma):
         )
 
 
+class InterpolatedRV(RandomVariable):
+    name = "interpolated"
+    ndim_supp = 0
+    ndims_params = [1, 1]
+    dtype = "floatX"
+    _print_name = ("Interpolated", "\\operatorname{Interpolated}")
+
+    @classmethod
+    def rng_fn(cls, rng, x, pdf, size=None):
+        interp = InterpolatedUnivariateSpline(x, pdf, k=1, ext="zeros")
+
+        Z = interp.integral(x[0], x[-1])
```

Comment:
Inside `rng_fn` there shouldn't be any symbolic operations. I didn't realize you need `Z` here as well.

You can either use scipy equivalents or, if easier, have everything that's needed for both `logp` and the `rng_fn` defined once during the `dist classmethod` and passed to `super` as additional inputs (and expected in both the `logp` and `rng_fn`. This way, the `rng_fn` will receive the non-symbolic arguments as inputs during normal sampling, while the `logp` method will receive the `symbolic` ones. 

Apologies if this sounds very confusing :D

Reply:
I think I should probably refactor the test first, just so that I'll have an intuition of how this entire `Interpolated` thing should work.

Reply:
Yeah, you can also check it out in V3, and see how it works there (and in the test). That may help, since this is a very non-standard one